lib/dry/types/hash/schema.rb

```diff
@@ -149,8 +149,6 @@ def resolve(hash)
           end
         end
       end
-
-      private_constant(*constants(false))
```

Comment:
Well why do we make methods private instead of just documenting them as private API? In both cases it is a signal to other collaborators and to end users that the method (or constant) is an implementation detail that shouldn't be interacted with directly.